### ISSUE TITLE
unused_braces warning emitted for braces that have semantic value

### ISSUE THREAD

**Issue by cartercanedy** [rust-lang/rust#154247]
### Summary

Since the adoption of [tail expression temporary scope changes](https://doc.rust-lang.org/edition-guide/rust-2024/temporary-tail-expr-scope.html) in the 2024 edition, it is safe to wrap a method call on a temporary lock guard in a block with assurance that the lock will be dropped before exiting the scope. It seems like this lint is either being emitted in error (false positive) or the other, very similar expression in the same file is not being detected (false negative).

### Lint Name

unused_braces

### Reproducer

Diagnostic emitted by rustc:
```
warning: unnecessary braces around method argument
   --> .../file.rs:459:29
    |
459 |         self.update({ self.lockable.lock().do_work()?.collect() })
    |                     ^^                                         ^^
    |
    = note: `#[warn(unused_braces)]` on by default
help: remove these braces
    |
459 -         self.update({ self.lockable.lock().do_work()?.collect() })
459 +         self.update(self.lockable.lock().do_work()?.collect())
```

There's a similar implementation in an impl block right below this that does not have this lint emitted

### Version

```text
rustc 1.96.0-nightly (562dee482 2026-03-21)
binary: rustc
commit-hash: 562dee4820c458d823175268e41601d4c060588a
commit-date: 2026-03-21
host: aarch64-apple-darwin
release: 1.96.0-nightly
LLVM version: 22.1.0
```
